src/Widgets/AbstractAppContainer.vala

```diff
@@ -73,7 +73,18 @@ namespace AppCenter {
                     return false;
                 }
 
-                return this.package.get_payments_key() != null;
+                return this.package.get_payments_key () != null;
+            }
+        }
+
+        private string suggested_amount_;
+        public string suggested_amount {
+            get {
```

Comment:
I don't think it is ever needed to check if package is `null`. If it could, then the package would be nullable. Also, all packages must have their component, since they can't be constructed without it, so this check is also not needed here. The Package class also takes care of handling when there is no such key, so you can also remove returning `DEFAULT_PRICE_DOLLARS`. What I would like to see here is just:

`public string suggested_amount {
   get {
             return package.get_suggested_amount ();
   }
}`